diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a068e32..19216b0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -80,6 +80,7 @@ endif()
 
 # Define flags with cmake instead of by defining them on win32_config.h
 if(WITHOUT_WIN32_CONFIG)
+  list(APPEND rdkafka_compile_definitions WITHOUT_WIN32_CONFIG)
   if(WITH_SSL)
     list(APPEND rdkafka_compile_definitions WITH_SSL)
   endif(WITH_SSL)
@@ -117,10 +118,10 @@ target_compile_definitions(rdkafka PUBLIC ${rdkafka_compile_definitions})
 if(RDKAFKA_BUILD_STATIC)
   target_compile_definitions(rdkafka PUBLIC LIBRDKAFKA_STATICLIB)
 endif()
-if(WIN32)    
+if(WIN32)
   target_link_libraries(rdkafka PUBLIC crypt32)
   if(NOT RDKAFKA_BUILD_STATIC)
-    target_compile_definitions(rdkafka PRIVATE LIBRDKAFKA_EXPORTS)  
+    target_compile_definitions(rdkafka PRIVATE LIBRDKAFKA_EXPORTS)
   endif()
 endif()
 
